machines/vic/install.go

```diff
@@ -0,0 +1,309 @@
+		b, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		if err != nil {
+			log.Fatalf("Failed to read password from stdin: %s", err)
+		}
+		data.passwd = string(b)
+	}
+	data.conf.TargetPath = fmt.Sprintf("%s:%s@%s/sdk", data.user, data.passwd, data.target)
+}
+
+func validateConfiguration() error {
+	if err := loadCertificate(); err != nil {
+		return err
+	}
+
+	return data.conf.ValidateConfiguration()
+}
+
+func loadCertificate() error {
+	var keypair *Keypair
+	if data.cert != "" && data.key != "" {
+		log.Infof("Loading certificate/key pair - private key in %s", data.key)
+		keypair = NewKeyPair(false, data.key, data.cert)
+	} else if data.tlsGenerate {
+		data.key = fmt.Sprintf("./%s-key.pem", data.displayName)
+		data.cert = fmt.Sprintf("./%s-cert.pem", data.displayName)
+		log.Infof("Generating certificate/key pair - private key in %s", data.key)
+		keypair = NewKeyPair(true, data.key, data.cert)
+	}
+	if keypair == nil {
+		return nil
+	}
+	if err := keypair.GetCertificate(); err != nil {
+		log.Errorf("Failed to read/generate certificate: %s", err)
+		return err
+	}
+	data.conf.KeyPEM = keypair.KeyPEM
+	data.conf.CertPEM = keypair.CertPEM
+	return nil
+}
+
+func checkImagesFiles() error {
+	// detect images files
+	osImgs, ok := images[data.osType]
+	if !ok {
+		return fmt.Errorf("Specified OS \"%s\" is not known to this installer", data.osType)
+	}
+
+	var imgs []string
+	if data.applianceISO != "" {
+		imgs = append(imgs, data.applianceISO)
+	} else {
+		imgs = append(imgs, images["core"]...)
+	}
+	if data.bootstrapISO != "" {
+		imgs = append(imgs, data.bootstrapISO)
+	} else {
+		imgs = append(imgs, osImgs...)
+	}
+
+	for _, img := range imgs {
+		_, err := os.Stat(img)
+		if os.IsNotExist(err) {
+			var dir string
+			dir, err = filepath.Abs(filepath.Dir(os.Args[0]))
+			_, err = os.Stat(filepath.Join(dir, img))
+			if err == nil {
+				img = filepath.Join(dir, img)
+			}
+		}
+
+		if os.IsNotExist(err) {
+			log.Warnf("\t\tUnable to locate %s in the current or installer directory.", img)
+			return err
+		} else {
+			data.conf.ImageFiles = append(data.conf.ImageFiles, img)
+		}
+	}
+	return nil
+}
+
+func main() {
+	var err error
+	processParams()
+
+	if err = checkImagesFiles(); err != nil {
+		log.Fatalf("%s", err)
+	}
+
+	// Open log file
+	f, err := os.OpenFile(data.logfile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+	if err != nil {
+		log.Fatalf("Error opening logfile %s: %v", data.logfile, err)
+	}
+	defer f.Close()
+
+	// Initiliaze logger with default TextFormatter
+	log.SetFormatter(&log.TextFormatter{ForceColors: true, FullTimestamp: true})
+	// SetOutput to io.MultiWriter so that we can log to stdout and a file
+	log.SetOutput(io.MultiWriter(os.Stdout, f))
+
+	log.Infof("### Installing VCH ####")
+
+	if err = validateConfiguration(); err != nil {
+		log.Fatalf("Validating supplied configuration failed. Exiting...")
+	}
+
+	executor := reification.NewDispatcher(data.conf, data.force)
```

Comment:
Seems that `reification` is not really used in the US - we should switch to a more easily comprehensible name. I'm actually ok with `install`. While it's not entirely accurate people do have a good idea what it means and it shouldn't introduce too much confusion if reconfiguration is accomplished in the same package.